### ISSUE TITLE
Contributing Tests

### ISSUE THREAD

**Issue by jakelever** [DiseaseOntology/HumanDiseaseOntology#524]
Hi, some of the resources we build depend on the Disease Ontology so we've had a few problems with recent releases, related to the IS_A issues (e.g. #522). I noticed that there used to be tests by the Build Unknown badge on the Readme. Would it help if I contributed some tests to check the validity of some of the DO files? I could submit a Pull Request with some Python code that checks it (which could then be added to as needed). And it could run using Travis CI as the old tests did.

Let me know your thoughts,

All the best, Jake

**Comment by lschriml** [comment]
Hello Jake,
   Super !! 
I really appreciate your offer to help out.
We were testing out doing our release via ROBOT, and did run into a few
minor adjustments.

I just pushed out a release. 
We have tested these files, and hopefully found all of the glitches. 

It would be great to have you contribute tests for the validity of the DO files. 

Much appreciated,
Lynn

**Comment by jakelever** [comment]
Hey @lschriml . Great! I'm not 100% of what the experted properties of the different files are. The tests I had in mind are below. Please correct me. If you have any other ideas, I'd happily implement them.

- Parent Properties
  - Check that all items in doid.obo have at least one parent
  - Check that all items in doid-non-classified.obo have exactly one parent
- Connectedness
  - Check that all items in doid.obo are connected through IS_A properties
  - Check that all items in doid-non-classified.obo are connected through IS_A properties
- Tree Structure
  - Check that the overall structure of doid-non-classified.obo is a tree with disease at the root

**Comment by lschriml** [comment]
Super, thank you Jake !!

**Comment by rctauber** [comment]
Hi @jakelever - I'm working on writing your tests as SPARQL queries to run `robot verify` over. Thanks for contributing these ideas! `robot verify` can still be added for travis integration, so this will be very helpful.

I have the following checks:
1. dnc-verify-connectivity (runs only on non classified)
2. dnc-verify-single-parent (runs only on non classified)
3. verify-no-cycles (runs on doid.obo and non classified)
4. verify-no-orphans (runs on doid.obo and non classified)

The second one is the only one failing right now. `doid-edit` has a few asserted multiple parentages, which we are working on cleaning up with the logical defs. `doid-non-classified` is also generated after running the reasoner, but if we generate it *prior* to the reasoner, it wouldn't have the inferred multiple parents added in. Is that what is expected of this file?

**Comment by jakelever** [comment]
Great. I didn't realise you could robot for this.

**Comment by lschriml** [comment]
This is new for us. 
Becky is part of the robot development team, so we were able to implement 
(And test) the newest of robot in DO. 
It is a big improvement for us. 
I particularly like the flexibility of it. 
And that it also supports using SPARQL.

Cheers,
Lynn

**Comment by rctauber** [comment]
Tests are now added and I updated Travis. The current build can be found [here](https://travis-ci.org/DiseaseOntology/HumanDiseaseOntology).

I did include the single-parent check in the `src/sparql` directory, but that is not part of the tests right now since we still have multiple asserted parents. We can always add this in later. There are details in `src/sparql` in the README on how to add more queries.
